### DISCUSSION TITLE
How to download a blob file from JSON?

### DISCUSSION THREAD

**Discussion by indiedev84** [Budibase/budibase#11132] [Help] [answered]
Hello.
Used Budibase **[File Upload Tool](https://github.com/chungchunwang/Budibase-File-Upload)** plugin, which allows you to convert the file into base64 encoding and upload to the database. 
I can not download it back. 
Because the link returns the entire json
`https://domain.com/.[%7B%22name%22:%22%D0%A2%D0%9E%D0%A0%D0%93-12. %20%D0%A2%D0%BE%D0%B2%D0%B0%D1%80%D0%BD%D0%B0%D1%8F%20%D0%BD%D0%B0%D0%BA%D0%BB%D0%B0%D0%B4%D0%BD%D1%8B%D0%B5%20%D0%B7%D0%B0%20030723. pdf%22,%22type%22:%22application/pdf%22,%22size%22:191843,%22link%22:%22data:application/pdf;base64,JVBERi0xLjQNCiXi48/TDQoxIDAgb2JqDQo8PA0KL ...].`
How do I get the file to download?
Tried 
{{Submission Table.DS_FOR_USER.FileUrl.[0].link}} and 
{{Submission Table.DS_FOR_USER.FileUrl.0.link}}
Doesn't work, or do I need access to data? 
Can you tell me how to get it?

**Comment by melohagan** [comment]
Hey @indiedev84 

When you upload a file, the file opens a blob file in a new tab upon clicking:

![Screenshot 2023-07-14 at 12 42 46](https://github.com/Budibase/budibase/assets/101575380/a12fff62-617a-476a-bdfc-2f6ea0889bb6)

![Screenshot 2023-07-14 at 12 43 11](https://github.com/Budibase/budibase/assets/101575380/79097774-e12d-422f-837f-9a15ec1297cb)

The component does not appear to expose the blob URLs which would allow for custom file management outside of the component, but you would need to raise that on the file upload repository.

---

I was not able to find a way to readily use the `DataURL` due to Content Security Policy:

![Screenshot 2023-07-14 at 12 57 22](https://github.com/Budibase/budibase/assets/101575380/998366b5-e8ed-46c1-82f8-49227b316990)

Furthermore, because the base64 data is stored as a string, there is a bit of JavaScript to pull out the link:
```js
let length = $("New Repeater.Pets.Breed_test")?.length - 1;

return JSON.parse($("New Repeater.Pets.Breed_test")?.substring(1, length))[0].link;
```

**Reply by AlejandroFNadal** [reply, marked as answer]
I succesfully triggered a download from the DataURL using a navigate to action with the url, using the option "new tab"
